tests/ignite/contrib/metrics/test_precision_recall_curve.py

```diff
@@ -202,13 +202,15 @@ def _test(n_epochs, metric_device):
         metric_device = torch.device(metric_device)
         n_iters = 80
         size = 151
-        y_true = torch.randint(0, 2, (size,)).to(device)
-        y_preds = torch.randint(0, 2, (size,)).to(device)
+
+        offset = n_iters * size
+        y_true = torch.randint(0, 2, size=(offset * idist.get_world_size(),)).to(device)
+        y_preds = torch.randint(0, 2, size=(offset * idist.get_world_size(),)).to(device)
 
         def update(engine, i):
             return (
-                y_preds[i * size : (i + 1) * size],
-                y_true[i * size : (i + 1) * size],
+                y_preds[i * size + rank * offset : (i + 1) * size + rank * offset],
+                y_true[i * size + rank * offset : (i + 1) * size + rank * offset],
```

Comment:
Can you explain what that does exactly ?
Have you checked if generated data is different on each process ?

Reply:
Each process sets the same seed for random so each proc should have the same data (I suppose). However, each proc generates the data for itself and others procs, even if it's similar, each proc works on a different chunk of data. It means no sync over the procs needed to evaluate sklearn. 

Sync is replaced by data duplication (as far I understand).

Reply:
@sdesrozis I know what all that does, the question was to @sayantan1410 if he understands what he's doing :)

I agree that it would be better to make tests uniform: `_test_distrib_compute` and `_test_distrib_integration`. We have to open a new issue, explain the problem and send a follow-up PR for that.